### PR TITLE
ci: Add build and test CI

### LINKED ISSUE #4: Add CI

### PR DESCRIPTION
**Pull request by CohenArthur**
Closes #4

### CODE REVIEW

**Review by Skallwar — COMMENTED** [review]
Needs #3 to be merged before

**Comment by 08A** [comment]
@CohenArthur You can do:

ninja -C build (replace cd build; ninja)
and ninja -C build test

**Comment by 08A** [comment]
use as docker image alexisehret/archlinux_c_meson:latest

**Comment by Skallwar** [comment]
> use as docker image alexisehret/archlinux_c_meson:latest

Docker is too slow on Github Action

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build ubsh
+        run: |
+          sudo apt-get install meson ninja-build
```
Comment (Skallwar) [inline .github/workflows/build_and_test.yml:13]:
```suggestion
          apt install meson ninja-build
```

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  tests:
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run Unit Tests
+        run: |
+          sudo apt-get install meson ninja-build
```
Comment (Skallwar) [inline .github/workflows/build_and_test.yml:26]:
```suggestion
          apt install meson ninja-build
```

**Comment by 08A** [comment]
@Skallwar ubuntu is not a good target platform, meson is too old and some lib are outdated

**Comment by Skallwar** [comment]
> @Skallwar ubuntu is not a good target platform, meson is too old and some lib are outdated

@08A Archlinux is not stable. The last thing we want is a broken CI

**Comment by CohenArthur** [comment]
Docker is way too slow on github actions, I agree. Especially for a project this size, and where special libraries are not "needed" (yet)

**Comment by 08A** [comment]
First: we don't need a fast ci. Because before avery we need to review changes.

Second: with a docker image you have always a stable env. So "Archlinux is not stable." is not a valid argument

**Comment by Skallwar** [comment]
> So "Archlinux is not stable." is not a valid argument

So your complaint about Ubuntu not being up to date is irrelevant if you freeze your docker image to make it stable

**Comment by CohenArthur** [comment]
> Needs #3 to be merged before

Also I think we should merge this before #3 , in order to already have a CI in place for #3. Sure, the CI's PR won't pass the CI, but I feel like that's okay :D

**Review by Skallwar — APPROVED** [review]

**Review by n1tram1 — APPROVED** [review]

**Pull request merged**